packages/dx-chart-core/src/plugins/scale/computeds.ts

```diff
@@ -3,125 +3,122 @@ import { scaleLinear as d3ScaleLinear, scaleBand as d3ScaleBand } from 'd3-scale
-  return scaleLinear;
+  const factory = guessFactory(series.points, getItem);
+  return {
+    ...domain,
+    factory,
+    isDiscrete: isDiscrete(factory),
+  };
 };
 
-const collectDomainsFromSeries = (domains: DomainInfoCache, seriesList: SeriesList) => {
-  seriesList.forEach((seriesItem) => {
-    if (!domains[ARGUMENT_DOMAIN].factory) {
-      Object.assign(domains[ARGUMENT_DOMAIN], {
-        factory: guessFactory(seriesItem.points, getArgument),
-      });
-    }
-    const valueDomainName = getSeriesValueDomainName(seriesItem);
-    const obj = domains[valueDomainName];
-    if (!obj.factory) {
-      obj.factory = guessFactory(seriesItem.points, getValue);
-    }
-    // TODO: It is to be removed together with *TODO* from above.
-    if (seriesItem.getPointTransformer.isStartedFromZero && obj.domain.length === 0) {
-      obj.domain = [0];
-    }
-  });
-  Object.keys(domains).forEach((name) => {
-    const obj = domains[name];
-    if (!obj.factory) {
-      obj.factory = scaleLinear;
-    }
-    obj.isDiscrete = !!obj.factory().bandwidth;
-  });
-  return domains;
+const updateDomainItems = (domain: DomainInfo, items: DomainItems): DomainInfo => {
+  const merge = domain.isDiscrete ? mergeDiscreteDomains : mergeContinuousDomains;
+  const merged = merge(domain.domain, items);
+  return merged === domain.domain ? domain : {
+    ...domain,
+    domain: domain.modifyDomain ? domain.modifyDomain(merged) : merged,
+  };
 };
 
-const customizeDomains = (domains: DomainInfoCache) => {
-  Object.keys(domains).forEach((name) => {
-    const obj = domains[name];
-    if (obj.modifyDomain) {
-      obj.domain = obj.modifyDomain(obj.domain);
-    }
-  });
+const getArgumentDomainItems: GetDomainItemsFn = series => series.points.map(getArgument);
+
+const getValueDomainItems: GetDomainItemsFn = (series) => {
+  // TODO: This is a temporary workaround for Stack plugin.
+  // Once scales (or domains) are exposed for modification Stack will modify scale and
+  // this code will be removed.
+  const items = series.getValueDomain
+    ? series.getValueDomain(series.points) : series.points.map(getValue);
+  // TODO: It is also a part of that workaround.
+  return series.getPointTransformer.isStartedFromZero ? [0, ...items] : items;
 };
 
+const updateDomain = (
+  domain: DomainInfo, series: Series, getItem: GetItemFn, getDomainItems: GetDomainItemsFn,
+) => updateDomainItems(updateDomainFactory(domain, series, getItem), getDomainItems(series));
+
 /** @internal */
-export const computeDomains: ComputeDomainsFn = (domains, seriesList) => {
-  const result = copy(domains);
-  collectDomainsFromSeries(result, seriesList);
-  calculateDomains(result, seriesList);
-  customizeDomains(result);
-  return result;
+export const extendDomains: ExtendDomainsFn = (domains, series) => {
+  const argumentDomain = updateDomain(
+    domains[ARGUMENT_DOMAIN], series, getArgument, getArgumentDomainItems);
+  const valueDomainName = getSeriesValueDomainName(series);
```

Comment:
Why is `getValueDomainName` not called directly here?

Reply:
`getSeriesValueDomainName` is removed.